salt/states/pkg.py

```diff
@@ -2329,12 +2329,16 @@ def latest(
     '''
     refresh = salt.utils.pkg.check_refresh(__opts__, refresh)
 
+    if not name:
+        # Note state checks name is a string, this should only happen if their is a bug
+        raise CommandExecutionError('Parameter name must contain label or package name.')
+
     if kwargs.get('sources'):
         return {'name': name,
                 'changes': {},
                 'result': False,
                 'comment': 'The "sources" parameter is not supported.'}
-    elif pkgs:
+    elif pkgs is not None:
```

Comment:
Again, why would this be necessary? We shouldn't be invoking this state with an empty list as input, so why do we need to specifically check if it is not None?